rest/admin_api.go

```diff
@@ -915,19 +915,32 @@ func (h *handler) handlePutDbConfigImportFilter() error {
 // In non-persistent mode, the endpoint just removes the database from the node.
 func (h *handler) handleDeleteDB() error {
 	h.assertAdminOnly()
+	dbName := h.PathVar("olddb")
+
+	var bucket string
+
+	dbContext, _ := h.server.GetDatabase(h.ctx(), dbName)
+
+	if dbContext != nil {
+		bucket = dbContext.Bucket.GetName()
+	} else if h.server.persistentConfig {
+		_, cnf, err := h.server.fetchDatabase(h.ctx(), dbName)
+		bucket = *cnf.Bucket
```

Comment:
If we got an error `cnf` will probably be nil and this will panic.

I think it's just a case of moving error handling up to avoid?

Reply:
Addressed in https://github.com/couchbase/sync_gateway/pull/5894/commits/e73f4d4d4f01b3a447bab6ebc4695715fd207418